samples/gibbs_ensemble/gibbs_ensemble_socket.py

```diff
@@ -443,10 +442,11 @@ def check_exchange_particle(source_box, dest_box, inner_potential):
         # timing
         tock = time.time()
 
-        print("step %d, densities %.3f %.3f, volumes %.2f, %.2f %.f ms / move" % 
-              (i, densities[0][-1], densities[1][-1], 
-               boxes[0].box_l**3, boxes[1].box_l**3,
-               (tock - tick) * 1000 / REPORT_INTERVAL))
+        if len(densities[0]):
+            print("step %d, densities %.3f %.3f, volumes %.2f, %.2f %.f ms / move" % 
```

Comment:
this snippet gets a lot more readable if you use format strings ala ```print(f'step {i})```. I also think that in general it's good practice to *not* directly print stuff to stdout but instead use the logging infrastructure of python.